diag_manager/fms_diag_file_object.F90

```diff
@@ -0,0 +1,407 @@
+  has_var_ids = allocated(obj%var_ids)
+end function has_var_ids
+!> \brief Logical function to determine if the variable var_list has been allocated or associated
+!! \return .True. if var_list exists .False. if var_list has not been set
+pure logical function has_var_list (obj)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  has_var_list = allocated(obj%var_list)
+end function has_var_list
+!> \brief Returns a copy of the value of id
+!! \return A copy of id
+pure function get_id (obj) result (res)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  integer :: res
+  res = obj%id
+end function get_id
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! TODO
+!> \brief Returns a copy of the value of fileobj
+!! \return A copy of fileobj
+!pure function get_fileobj (obj) result (res)
+!  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+!  class(FmsNetcdfFile_t) :: res
+!  res = obj%fileobj
+!end function get_fileobj
+!> \brief Returns a copy of the value of file_domain_type
+!! \return A copy of file_domain_type
+pure function get_file_domain_type (obj) result (res)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  character(1) :: res
+  res = obj%file_domain_type
+end function get_file_domain_type
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! TODO
+!!> \brief Returns a copy of the value of diag_yaml
+!!! \return A copy of diag_yaml
+!#ifdef use_yaml
+!pure function get_diag_yaml (obj) result (res)
+!  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+!  type(diagYamlFiles_type) :: res
+!  res = obj%diag_yaml
+!end function get_diag_yaml
+!#endif
+!> \brief Returns a copy of the value of file_metadata_from_model
+!! \return A copy of file_metadata_from_model
+pure function get_file_metadata_from_model (obj) result (res)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  character(len=:), dimension(:), allocatable :: res
+  res = obj%file_metadata_from_model
+end function get_file_metadata_from_model
+!> \brief Returns a copy of the value of var_list
+!! \return A copy of var_list
+pure function get_var_list (obj) result (res)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  character(len=:), dimension(:), allocatable :: res
+!  allocate(character(len=len(obj%var_list(1))), dimension (size(obj%var_list(1))) :: res) 
+  res = obj%var_list
+end function get_var_list
+!> \brief Returns a copy of the value of var_ids
+!! \return A copy of var_ids
+pure function get_var_ids (obj) result (res)
+  class(fmsDiagFile_type), intent(in) :: obj !< The file object
+  integer, dimension(:), allocatable :: res
+  allocate(res(size(obj%var_ids)))
+  res = obj%var_ids
+end function get_var_ids
+!!!!!!!!! Functions from diag_yaml_file
+
+!> \brief Returns a copy of file_fname from the yaml object
+!! \return Copy of file_fname
+pure function get_file_fname (obj) result(res)
```

Comment:
If this is changed to get the diag_file part of the yaml (diag_yaml%diag_files(:)) instead of all of the diag_yaml. This can just be simplified to
`obj%diag_file%get_file_fname`

Reply:
It's just a wrapper for calling the yaml stuff.  It's mostly for convenience